draft-ietf-tls-esni.md

```diff
@@ -712,10 +712,11 @@ structure available for the server, it SHOULD send a GREASE {{?RFC8701}}
   according to {{padding}}.
 
 When sending a second ClientHello in response to a HelloRetryRequest, the
-client copies the `cipher_suite` field from the first ClientHello. It sets
-`config_id`, and `enc` to the empty string. Finally, it generates a new
-`payload` field, using the length of a padded second EncodedClientHelloInner
-for L.
+client copies the entire "encrypted_client_hello" extension from the first
+ClientHello.
+
+[[OPEN ISSUE: The above doesn't match HRR handling for either ECH acceptance or
+rejection. See issue #358.]]
```

Comment:
I'm confused by the word "match" here. Do you mean to say that greased ECH looks different from real ECH after HRR?

Reply:
Yes. This is #358. There is a potential compatibility issue with changing CH2's ECH extension when talking to a server that doesn't expect ECH. We also need to change CH2's ECH extension when actually speaking ECH.

This PR resolves the compatibility issue, just so we can get this out of the way of draft-09. It does not resolve the bulk of #358 which interacts with every other HRR problem.

Reply:
I think this is a good idea for ECH-09, I'm merely suggesting that the wording of the OPEN ISSUE could be clearer.

Reply:
Ah. What wording would you prefer? Perhaps: "OPEN ISSUE: The extension value is not copied when sending a real ECH extension. See issue #358."

Reply:
That's better, but I think we should make this an explicit don't-stick-out issue. Slight revision: "OPEN ISSUE: The extension value is not copied when sending a real ECH extension, causing this code path to stick out. See issue #358."

Reply:
It's not just about sticking out. ECH acceptance and ECH rejection currently also violate the RFC8446 requirement. This PR doesn't change it because it would break them, and it's mostly okay compatibility-wise because it would only affect servers that have opted in via DNS. (But only mostly. The recovery flow on a strictly enforcing server is broken.)

Reply:
That's why I specifically wanted this to just be a quick summary and then a pointer to the issue where we'll capture the actual discussion in a more fluid form.

Reply:
Yeah, I think the comment is fine as-is.